Readme.md

```diff
@@ -39,20 +120,152 @@ kubectl apply/delete -f k8s-deployment.yaml
+
+The `Storage` must have a specific folder structure. All files have to be named after their **SHA256**. The folder structure consists of the first hex byte of the SHA256, which contains the second hex byte of the SHA256 as a sub-folder. In the sub-folder the sample itself is stored.
+
+```
+# Example of the expected sample structure
+/mnt/sampleexportstorage
+  - /00
+    - /00
+      - /00002455f77b5f921cb5eb23add0db9077769c0cacb840c460f41d9a8357929f
+      - ..
+    - /01
+    - ..
+    - /FF
+  - /01
+  - ..
+  - /FF
+```
+
+For the configuration of users and their corresponding data sets, the [shareconfig.yaml](./src/SampleExchangeApi.Console/sh
+) is used. The MongoDB does not know about any users, it only contains samples which belong to a set.
+
+```yaml
+# Example sharedconfig.yaml with two exchange partners
+Partners:
+- Name: partner1 # Name of the exchange partner
+  Password: 466fef588adae318d7f50541982785daaf61d51b5c47101c1c751fbd717dd9e8 # Password Hash
+  Salt: 79b48cd1d1ed8fa129c58c5c2d0633b3f9d46087feb8b0165a5ed560356db894 # Password Salt
+  Enabled: Yes # Is the exchange with the partner enabled?
+  Sampleset: Classic # Which set it shared with the partner?
+
+- Name: partner2
+  Password: c5363549da9f03d8da44db70ec12ca5dce8078d4cb5fda1d7ecadd4372031539
+  Salt: 8ec1690da1bf1baad62a20c0db8e4ad26205ec577b741ccc8b1e2e834670a5e4
+  Enabled: No
+  Sampleset: Extended
+```
+
+The [main.py](./src/FillMongoWithTestData/main.py) is an example script, which show how the MongoDB is filled with samples to share. It does two things. First it moved the sample itself to the sample folder, as described above. Second, it inserts the needed meta-data for the sample into the MongoDB. This is all that is needed to be able to share the sample with a partner.
 
- By executing ```curl -u "testuser:somenicepassword" -X GET -k -i 'http://localhost:8080/v1/list?start=2020-09-23'``` you will receive
- a list of JWT tokens, which generally have the format ```aaaa.bbbbbbbb.cccc``` of three base64 encoded sections that are separated
- by dots. The first section are header information, declaring the structure as JWT and the used hash algorithm. The second part is the
- actual payload that contains expiration date of the token, SHA256, file size and the platform. The third part is a signature that guarantees that the JWT is valid.
- If after checking that the sample is not already part of your collection and you have interest in the reported platform,
- you can download it with ```curl -X GET -k "http://localhost:8080/v1/download?token=$PUT_TOKEN_HERE"```.
+```python
+#!/usr/bin/python3
+
+import hashlib
+import pymongo # sudo pip install pymongo
+import datetime
+import os
+import sys, getopt
+
+def put_string_into_db(sha256, platform, file_size, sample_set, mongo_collection):
+    current_iso_datetime = datetime.datetime.utcnow()
+    entry = {
+                "_id": f"{sha256}:test",                  # Unique ID 
```

Comment:
I'm not sure where the ":test" comes from. Does somebody know, why it's needed and what value should be set here?

Reply:
This stands for the sample-set. If the test-data is just filled with this one sample-set you could also just remove it. If you leave it there, maybe rename it to testSampleSet

Reply:
So shouldn't it be the same value as the "SampleSet" property below?